lib/reline/line_editor.rb

```diff
@@ -191,29 +199,29 @@ def handle_signal
     Reline::IOGate.move_cursor_column 0
     @rendered_screen.lines = []
     @rendered_screen.cursor_y = 0
-    case @old_trap
+    case @old_int_trap
     when 'DEFAULT', 'SYSTEM_DEFAULT'
       raise Interrupt
     when 'IGNORE'
       # Do nothing
     when 'EXIT'
       exit
     else
-      @old_trap.call if @old_trap.respond_to?(:call)
+      @old_int_trap.call if @old_int_trap.respond_to?(:call)
     end
   end
 
   def set_signal_handlers
-    Reline::IOGate.set_winch_handler do
+    @old_winch_trap = Reline::IOGate.set_winch_handler do
```

Comment:
`Reline::IOGate.set_winch_handler` does not return `old_winch_trap`.

`Reline::ANSI` returns SIGCONT trap
`Reline::Windows` returns new_winch_trap
`Reline::Dumb` returns nil

I think we should do `old_winch_trap.call` inside `lib/reline/io/ansi.rb`.
```ruby
def set_winch_handler(&handler)
  @old_winch_handler = Signal.trap('WINCH') do
    # do it here?
  end
  @old_cont_handler = Signal.trap('CONT') do
    # maybe we can fix the same problem with SIGCONT here
  end
end
```

Reply:
Ah, great point. I've only been using the ANSI adapter, so didn't think to check the others and what they return. Will think on how to shift this down one level